.github/workflows/build_rocks.yaml

```diff
@@ -61,32 +61,29 @@ jobs:
         uses: canonical/craft-actions/rockcraft-pack@main
         with:
           path: ${{ matrix.path }}
-      - name: Upload rock to ghcr.io
-        if: ${{ !github.event.pull_request.head.repo.fork }}
+      - name: Extract rock information
         run: |
           IMAGE_NAME=$(yq '.name' ${{ matrix.path }}/rockcraft.yaml)
           IMAGE_REF=${{ inputs.registry }}/${{ inputs.owner }}/$IMAGE_NAME:${{ github.run_id }}
-          skopeo --insecure-policy copy oci-archive:$(ls ${{ matrix.path }}/${{ matrix.image }}*.rock) docker://$IMAGE_REF --dest-creds "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}"
           echo "IMAGE_NAME=$IMAGE_NAME" >> $GITHUB_ENV
           echo "IMAGE_REF=$IMAGE_REF" >> $GITHUB_ENV
+      - name: Upload rock to ghcr.io
+        if: ${{ !github.event.pull_request.head.repo.fork }}
```

Comment:
Are there any problems with not uploading the rock if the run is from a fork? What would we use that for?

Reply:
When running from a fork, the Github action does not have sufficient permission to push container images to the ghcr registry.

Reply:
And what are the consequences of not having the image in the ghcr registry? Is it used later on in the workflow?

Reply:
I think in such case the image is using a locally build image, which is then pushed to a local registry in microk8s therefore ghcr repository is not needed.
The final image will anyway be built and pushed to ghcr when the merge on main will occur

Reply:
In a PR from a forked repository, the image is uploaded as an artifact instead of push to ghcr registry. And in the following integration tests, the image is downloaded from artifacts or pull from ghcr registry according to the initiator of the PR.

Reply:
If the image was pushed to the microk8s registry, it will be gone once the `build_rocks` job finished.